hack/update-codecgen.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
```

Comment:
You're both right. The key to this is that `$BASH_SOURCE` here is the elided (unreadable) form of `$BASH_SOURCE[0]`, e.g.:

```
zml@zml:~/kubernetes$ FOO=(1 2 3)
zml@zml:~/kubernetes$ echo $FOO
1
zml@zml:~/kubernetes$ echo ${FOO[0]}
1
```